content/blog/2022/2022-08-03-feature-flags-in-nodejs-backend-server-with-launchdarkly.md

```diff
@@ -0,0 +1,720 @@
+		}
+ 
+		// Given the minimum log level, determine if the level in question can be logged.
+		switch ( this.previousLevel = minLogLevel ) {
+			case 'error':
+				return level === 'error';
+			case 'warn':
+				return level === 'error' ||	level === 'warn';
+			case 'info':
+				return level === 'error' || level === 'warn' || level === 'info';
+			default:
+				return true;
+		} 
+	} 
+}
+
+export default LdLogger;
+```
+
+Next we will define the logic to subscribe to this log-level and execute some operations in a loop. For this testing, we can simply define a fake memory reader that will generate some random number and print message accordingly:
+
+```javascript
+import chalk from 'chalk';
+import LaunchDarkly from 'launchdarkly-node-server-sdk';
+import LdLogger from './ld_logger.js';
+
+const LD_SDK_KEY = 'sdk-d2432dc7-e56a-458b-9f93-0361af47d578';
+const flagKey = 'backend-log-level';
+const buyerKey = 'mob-b9d6d4d4-4300-46fa-9b13-d9eac89f9794';
+const launchDarklyClient = LaunchDarkly.init( LD_SDK_KEY );
+launchDarklyClient.waitForInitialization();
+const logger = new LdLogger( launchDarklyClient, flagKey, buyerKey );
+let loop = 0;
+ 
+launchDarklyClient.once('ready', async () => { 
+		setTimeout( executeLoop, 1000 ); 
+	}
+);
+ 
+//Fake memory reader randomized to throw errors.
+function readMemory() { 
+	const memory = ( Math.random() * 100 ).toFixed( 1 ); 
+	if ( memory <= 30 ) { 
+		throw new Error( 'IOError' ); 
+	} 
+	return memory; 
+}
+
+function executeLoop () {
+		console.log( chalk.dim.italic( `Loop ${ ++loop }` ) ); 
+		logger.debug( 'Executing loop.' ); 
+		try { 
+			logger.debug( 'Checking free memory.' );
+			const memoryUsed = readMemory();
+			logger.info( `Memory used: ${ memoryUsed }%` ); 
+			if ( memoryUsed >= 50 ) { 
+				logger.warn( 'More than half of free memory has been allocated.' ); 
+			} 
+		} catch ( error ) { 
+			logger.error( `Memory could not be read: ${ error.message }` );
+		} 
+		setTimeout( executeLoop, 1000 ); 
+}
+```
+
+Next we can define the script as part of `package.json`:
+
+```json
+{
+	"scripts": {
+    	"dynamic": "node dynamic_logging.js"
+  	}
+}
+```
+
+Then we can execute the following command to run our app:
+
+```bash
+npm run dynamic
+```
+
+Finally when we run the above command it will print something like below:
+
+```bash
+info: [LaunchDarkly] Initializing stream processor to receive feature flag updates
+info: [LaunchDarkly] Opened LaunchDarkly stream connection
+Loop 1
+Switching to log-level: debug
+DEBUG: Executing loop.
+DEBUG: Checking free memory.
+INFO: Memory used: 68.2%
+WARN: More than half of free memory has been allocated.
+Loop 2
+DEBUG: Executing loop.
+DEBUG: Checking free memory.
+ERROR: Memory could not be read: IOError
+Loop 3
+DEBUG: Executing loop.
+DEBUG: Checking free memory.
+INFO: Memory used: 67.4%
+WARN: More than half of free memory has been allocated.
+```
+
+### Manage blacklisted Users or IPs on the Fly
```

Comment:
"Blacklist" is not considered an inclusive word. Better use "block users" or something similar. Can you change the places where you have used this word, please? Maybe use "blocklist" in the code.

```suggestion
### Block Users based on their IP-Address
```